android/src/main/java/com/magnet/mmx/client/api/MMXMessage.java

```diff
@@ -761,17 +814,35 @@ static MMXMessage fromMMXMessage(MMXTopic topic, com.magnet.mmx.client.common.MM
     //populate the message content
     HashMap<String, String> content = new HashMap<String, String>();
     for (Map.Entry<String,String> entry : message.getPayload().getAllMetaData().entrySet()) {
-      content.put(entry.getKey(), entry.getValue());
+      if(!CONTENT_ATTACHMENTS.equals(entry.getKey())) {
+        content.put(entry.getKey(), entry.getValue());
+      }
     }
 
-    MMXMessage newMessage = new MMXMessage();
+    MMXMessage.Builder newMessage = new MMXMessage.Builder();
+
+    // Extract attachments
+    String attachmentsStr = MagnetUtils.trimQuotes(message.getPayload().getAllMetaData().get(CONTENT_ATTACHMENTS));
+    if(StringUtil.isNotEmpty(attachmentsStr)) {
+      List<Attachment> attachments = GsonData.getGson().fromJson(attachmentsStr, new TypeToken<List<Attachment>>() {}.getType());
+      if(null != attachments && attachments.size() > 0) {
+        newMessage.attachments(attachments.toArray(new Attachment[0]));
+      }
+    }
+    Log.d(TAG, "-----------message convertion, topic : " + topic + ", message : " + message);
+    if(null != topic) {
```

Comment:
Hi @vincentwlau , line 833-841 is the fix for the bug (shouldn't set both channel and recipients for message). Please review.